### PR TITLE
[21424] Add test for support of enumeration literal `@value` annotation

### PR DESCRIPTION
**Pull request by richiware**
## Description

Add a new test for new enumeration in `dds-types-test` repository.

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- *N/A* Any new/modified methods have been properly documented using Doxygen. 
- *N/A* Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- [x] Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- [x] Changes are API compatible. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 
    
- *N/A* Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- _N/A_: If this is a critical bug fix, backports to the critical-only supported branches have been requested.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: thirdparty/fastcdr
Comment (MiguelCompany) [inline thirdparty/fastcdr:1]: This might need a rebase?

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp [hunk 1]
```diff
@@ -2111,7 +2111,8 @@ ReturnCode_t TypeObjectRegistry::register_typeobject_w_enum_dynamic_type(
                                          member_descriptor.is_default_label())};
         // TODO(richi): Literal value might be automatically assigned or taken from default_value (@value annotation)
```
[the remark this answers is not in the document]
Comment (richiware) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp]: Done in 28e8a02cc

**Review by MiguelCompany — APPROVED** [review]

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Pull request merged**